### DISCUSSION TITLE
2024-08-30 온라인 회의

### DISCUSSION THREAD

**Discussion by jimin52** [jiphyeonjeon-42/backend-nest#1] [General]
참여자 : jimin, yena, minsepar

1. 지난 시간 목표설정 중간점검
2. 이슈 공유
3. 프로젝트 전체기간 설정

**Comment by nyj001012** [comment]
## 진행도
### 백업 DB @jimin52 님께 전달
- 29일 완료

### 로컬 개발환경 구축
- 맥북으로 진행할 예정

## 목표
- 기존 api 하나 service, repository, swagger까지 nestjs로 변환하여 PR 올리기

**Comment by jimin52** [comment]
## 진행도
### DevContainer 만들어 공유하기
- 못함
### nestjs boilerplate 만들어 공유하기
- 못함. 예나님이 맡아주심
### 깃허브 프로젝트 만들기
- 함
### 디코 서버 초대하기
- 함

## 목표
- 주말까지 회사 업무 마치기
- 주말까지 devcontainer 설정하기

**Comment by parkj12b** [comment]
## 진행도
### nestJS 주요 개념공부하기

- nestjs 주요 개념 공부하기
- 의존성 주입
- module, provider, controller
- pipes, guard, intercepter, decorator
- validation, DTO, ORM
- typeorm migration
- 모듈간에 순환참조
- authorization authentication 차이
- REST api 원칙

### 아직 못함

- nestjs에서 crud 해보기 (tutorial)
- validation 은 nestjs-zod

### 목표

- 아직 못함 완료하기
- 자바스크립트 공부
- 타입스크립트 공부
- JS & TS 는 roadmap.sh

**Comment by jimin52** [comment]
### 이슈 공유
- yena: 이슈 없음
- jimin: 회사일 이번주까지 바쁨
- minsepar: 웹서브 이슈 (다음주 주말까지)

**Comment by jimin52** [comment]
## 앞으로 프로젝트 진행 방법
- 2주에 한번씩 sprint 계획
- 회의때마다 진행사항 확인, 이슈 공유
- 코드리뷰는 풀리퀘스트 올라오고 바로
- 만약 회의때까지 안되어있으면 라이브로 코드리뷰
